cpp/oneapi/dal/algo/dbscan/test/spmd_backend_template.hpp

```diff
@@ -0,0 +1,385 @@
+
+    using float_t = std::tuple_element_t<0, TestType>;
+
+    constexpr float_t data[] = { 0.0, 2.0, 3.0, 4.0, 6.0, 8.0, 10.0 };
+    const auto x = homogen_table::wrap(data, 7, 1);
+
+    constexpr double epsilon = 1;
+    constexpr std::int64_t min_observations = 3;
+
+    constexpr std::int32_t responses[] = { -1, 0, 0, 0, -1, -1, -1 };
+    const auto r = homogen_table::wrap(responses, 7, 1);
+
+    this->run_spmd_response_checks(x, table{}, epsilon, min_observations, r);
+}
+
+TEMPLATE_LIST_TEST_M(dbscan_spmd_backend_test,
+                     "dbscan simple core observations test #4",
+                     "[dbscan][batch]",
+                     dbscan_types) {
+    SKIP_IF(this->not_float64_friendly());
+    SKIP_IF(this->get_policy().is_cpu());
+
+    using float_t = std::tuple_element_t<0, TestType>;
+
+    constexpr float_t data[] = { 0.0, 2.0, 3.0, 4.0, 6.0, 8.0, 10.0 };
+    const auto x = homogen_table::wrap(data, 7, 1);
+
+    constexpr double epsilon = 1;
+    constexpr std::int64_t min_observations = 4;
+
+    constexpr std::int32_t responses[] = { -1, -1, -1, -1, -1, -1, -1 };
+    const auto r = homogen_table::wrap(responses, 7, 1);
+
+    this->run_spmd_response_checks(x, table{}, epsilon, min_observations, r);
+}
+
+TEMPLATE_LIST_TEST_M(dbscan_spmd_backend_test,
+                     "mnist: samples=10K, epsilon=1.7e3, min_observations=3",
+                     "[dbscan][nightly][batch][external-dataset]",
+                     dbscan_types) {
+    SKIP_IF(this->not_float64_friendly());
+    SKIP_IF(this->get_policy().is_cpu());
+    using float_t = std::tuple_element_t<0, TestType>;
+    constexpr bool is_double = std::is_same_v<float_t, double>;
+    // Skipped due to known issue
+    SKIP_IF(is_double);
+
+    const te::dataframe data =
+        te::dataframe_builder{ "workloads/mnist/dataset/mnist_test.csv" }.build();
+
+    const table x =
+        data.get_table(this->get_policy(), this->get_homogen_table_id(), sycl::usm::alloc::device);
+
+    constexpr double epsilon = 1.7e3;
+    constexpr std::int64_t min_observations = 3;
+    constexpr float_t ref_dbi = 1.584515;
+    this->run_spmd_dbi_checks(x, epsilon, min_observations, ref_dbi, 1.0e-3);
+}
+
+TEMPLATE_LIST_TEST_M(dbscan_spmd_backend_test,
+                     "hepmass: samples=10K, epsilon=5, min_observations=3",
+                     "[dbscan][nightly][batch][external-dataset]",
+                     dbscan_types) {
+    SKIP_IF(this->not_float64_friendly());
+    SKIP_IF(this->get_policy().is_cpu());
+    using float_t = std::tuple_element_t<0, TestType>;
+
+    const te::dataframe data = GENERATE_DATAFRAME(
+        te::dataframe_builder{ "workloads/hepmass/dataset/hepmass_10t_test.csv" });
+    const table x =
+        data.get_table(this->get_policy(), this->get_homogen_table_id(), sycl::usm::alloc::device);
+
+    constexpr double epsilon = 5;
+    constexpr std::int64_t min_observations = 3;
+    constexpr float_t ref_dbi = 0.78373;
```

Comment:
```suggestion
    constexpr float_t ref_dbi = float_t(0.78373);
```

Reply:
@makart19 - it will be implicitly converted in any case.